pythonforandroid/build.py

```diff
@@ -649,15 +644,17 @@ def run_pymodules_install(ctx, modules):
              'changes / workarounds.')
 
         # Make sure our build package dir is available, and the virtualenv
-        # site packages come FIRST (for the proper pip version):
+        # site packages come FIRST (so the proper pip version is used):
         env["PYTHONPATH"] += ":" + ctx.get_site_packages_dir()
         env["PYTHONPATH"] = os.path.abspath(join(
             ctx.build_dir, "venv", "lib",
             "python" + ctx.python_recipe.major_minor_version_string,
             "site-packages")) + ":" + env["PYTHONPATH"]
+
+        # Do actual install:
         shprint(sh.bash, '-c', (
-            "source venv/bin/activate && " +
-            "pip install -v --target '{0}' --no-deps -r requirements.txt"
+            "venv/bin/pip " +
```

Comment:
`source` sets paths for Python via the activating scripts. Why is the removal necessary? It doesn't seem correct to me.

Reply:
I suggested it because it works perfect and it's way more concise, but well you guys do as you want then

Reply:
Pip  can use the environment. If the environment is wrong, pip will behave wrong, that's why. :)

Reply:
I remember seeing someone mention recently that invoking the python binary inside the venv is the same as invoking the venv first then running python. Could it also be true for pip?

Reference here: https://news.ycombinator.com/item?id=18613891

Reply:
Yes I've always been doing this way and never had the issue the most import thing we're sourcing from the `activate` script is the `PATH` the rest is mostly playing with the `PS1`. But if @KeyWeeUsr prefers it the verbose way I'm also fine

Reply:
It looks like @tshirtman would also agree https://github.com/kivy/python-for-android/commit/8f1c25fac7c5aa24721d3e004b8f2dfaf74571f1 :stuck_out_tongue_winking_eye:

Reply:
@KeyWeeUsr I would *expect* that pip inside a venv would be wired to use the proper paths. However, I'm not 100% sure - but I would be very surprised if not

Reply:
@KeyWeeUsr this seems to suggest it's fine to just call pip directly: https://virtualenv.readthedocs.io/en/latest/userguide/#usage

(see how it says `This associated pip can be run from ENV/bin/pip.` without any remarks that this might be problematic)

Reply:
Ok, in that case, feel free to use the direct path. :)